app/src/main/java/com/matmoongi/data/RestaurantsRemoteDataSource.kt

```diff
@@ -1,15 +1,53 @@
 package com.matmoongi.data
 
+import android.annotation.SuppressLint
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.Priority
+import com.matmoongi.BuildConfig
+import com.matmoongi.LocationResultCallback
+import com.matmoongi.network.GoogleMapAPIService
+import com.matmoongi.response.Place
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.withContext
+import java.util.Locale
+
+private const val LAT_LNG_PARAMETER_FORMAT = "%.7f,%.7f"
 
 class RestaurantsRemoteDataSource(
+    private val googleMapAPIService: GoogleMapAPIService,
+    private val fusedLocationClient: FusedLocationProviderClient,
     private val ioDispatcher: CoroutineDispatcher,
 ) {
-    suspend fun getNearbyRestaurants(): List<Restaurant> =
-        // Move the execution to an IO-optimized thread since the ApiService
-        // doesn't support coroutines and makes synchronous requests.
+    /** 앱 권한 미동의 시 호출 불가 -> MainActivity에서 처리중 */
+    @SuppressLint("MissingPermission")
+    fun fetchCurrentLatLng(locationResultCallback: LocationResultCallback) {
```

Comment:
꼭 callback을 받아와야 하나요?
`Result<Coordinate>`을 반환하면 쉽게 해결되지 않을까요?

```suggestion
    suspend fun fetchCurrentGeoLocation(): Result<Coordinate> {
```

Reply:
Result 반환하도록 했습니다!